weave-js/src/common/components/elements/ModifiedDropdown.tsx

```diff
@@ -330,11 +330,10 @@ const ModifiedDropdown: FC<ModifiedDropdownProps> = React.memo(
         cursor: 'move',
       };
 
-      const labelText = item.text ?? '';
       const tagLabel = (
         <LabelTagContainer>
           <RemovableTag
-            label={labelText as string}
+            label={item.value?.toLocaleString() ?? ''}
```

Comment:
Can you explain the reason for switching item.text -> item.value and why we're doing toLocaleString?

Reply:
@ssisk 
This tag logic is within `renderLabel(item: DropdownItemProps, index: number, defaultLabelProps: LabelProps)` and then did a trace of the type for `item`

```
export interface DropdownItemProps extends StrictDropdownItemProps {
  [key: string]: any
}
export interface StrictDropdownItemProps {
  /** Display text. */
  text?: SemanticShorthandContent  <-- we were trying to display a reactNode which was causing the overflow issue

  /** Stored value. */
  value?: boolean | number | string <-- I saw this and felt this was a more sanitized value
}
export type SemanticShorthandContent = React.ReactNode
```

Inspect the element also confirmed that the `dropdown-with-icon` was causing issues in the Tailwind Tag comp
![Screenshot 2023-09-28 at 12 38 02 PM](https://github.com/wandb/weave/assets/103448723/67cf894e-fa93-4575-91d5-fdb37075ea97)

Then `RemovableTag` label field accepts string. Rather than trying to do another type assertion, `toString()` seemed like a good choice. I went a step further and thought that since `value` can also be a number, then `toLocaleString()` can nicely format our numbers
Example:
```
const num = 10000909932393
console.log(num.toLocaleString()) // returns "10,000,909,932,393" 
```

Reply:
hrm. I'm a little nervous about this change - 

toLocaleString() - that does seem nicer, but could have other side effects. Usually using the user's locale seems nice, but then they want to change it. I'd rather not change this without good reason to (and probably as part of a larger localization push)

switch text -> value: are we sure this is appropriate in all situations?